sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVDataSource.scala

```diff
@@ -155,7 +155,7 @@ object TextInputCSVDataSource extends CSVDataSource {
         sparkSession,
         paths = paths,
         className = classOf[TextFileFormat].getName,
-        options = options.parameters
+        options = options.parameters.originalMap
```

Comment:
this is not related to this PR, but I think this should be case sensitive for hadoop configs, so I am fixing it here.

Reply:
This is consistent with `DataFrameReader.loadV1Source`, +1